packages/compiler-cli/src/ngtsc/partial_evaluator/src/interpreter.ts

```diff
@@ -681,6 +688,45 @@ export class StaticInterpreter {
   private getReference<T extends DeclarationNode>(node: T, context: Context): Reference<T> {
     return new Reference(node, owningModule(context));
   }
+
+  private visitType(node: ts.TypeNode, context: Context): ResolvedValue {
+    if (ts.isLiteralTypeNode(node)) {
+      return this.visitLiteralType(node, context);
+    } else if (ts.isTupleTypeNode(node)) {
+      return this.visitTupleType(node, context);
+    } else if (ts.isNamedTupleMember(node)) {
+      return this.visitType(node.type, context);
+    }
+
+    return DynamicValue.fromDynamicType(node);
+  }
+
+  private visitLiteralType(node: ts.LiteralTypeNode, context: Context): ResolvedValue {
+    if (ts.isLiteralExpression(node.literal)) {
+      return this.visitExpression(node.literal, context);
+    }
+
+    switch (node.literal.kind) {
+      case ts.SyntaxKind.TrueKeyword:
+        return true;
+      case ts.SyntaxKind.FalseKeyword:
+        return false;
+      case ts.SyntaxKind.NullKeyword:
+        return null;
+      case ts.SyntaxKind.PrefixUnaryExpression:
+        return this.visitExpression(node.literal, context);
+    }
```

Comment:
Since primitive types are already handed in the `visitExpression` function [here](https://github.com/angular/angular/blob/master/packages/compiler-cli/src/ngtsc/partial_evaluator/src/interpreter.ts#L99-L107), we could just delegate the call? 
```suggestion
      case ts.SyntaxKind.TrueKeyword:
      case ts.SyntaxKind.FalseKeyword:
      case ts.SyntaxKind.NullKeyword:
      case ts.SyntaxKind.PrefixUnaryExpression:
        return this.visitExpression(node.literal, context);
    }
```

Reply:
yeah, good observation. I delete the entire `visitLiteralTypeNode` method since we can delete to `visitExpression`.